### DISCUSSION TITLE
Stopped with exit code 143

### DISCUSSION THREAD

**Discussion by Moosbueffel** [photoprism/photoprism#4071] [Q&A]
I have a installation at a Raspi 2GB RAM/4GB swap. PhotoPrism installed in a container. And the container is regular stopped. Start a manual scan will run without an error. Also there are no new pictures, or changed anything, so I don't know, why PhotoPrism start an update. But see the log of two exits:

`log:
time="2024-02-08T19:08:00Z" level=info msg="index: updated 1,021 photos [1m24.239163897s]"

time="2024-02-08T19:21:42Z" level=info msg="photo: 'Urlaub/Chiemsee 2004-08/100_0670' keeps existing manual title"

time="2024-02-08T19:21:45Z" level=info msg="photo: 'Urlaub/Chiemsee 2004-08/DSCF0039' keeps existing manual title"

time="2024-02-08T19:22:23Z" level=info msg="index: updated 473 photos [47.466242687s]"

time="2024-02-08T19:37:17Z" level=info msg="index: updated 175 photos [40.768562584s]"

---> Status:  Stopped for 13 hours with exit code 143

----------------------------------------------------
time="2024-02-16T11:08:43Z" level=info msg="index: updated 1,021 photos [1m26.294386679s]"

time="2024-02-16T11:22:23Z" level=info msg="photo: 'Urlaub/Chiemsee 2004-08/100_0670' keeps existing manual title"

time="2024-02-16T11:22:25Z" level=info msg="photo: 'Urlaub/Chiemsee 2004-08/DSCF0039' keeps existing manual title"

time="2024-02-16T11:23:04Z" level=info msg="index: updated 473 photos [47.832381251s]"

time="2024-02-16T11:37:57Z" level=info msg="index: updated 175 photos [41.115609127s]"

---> Status: Stopped for 19 hours with exit code 143`

I hope someone can help - thanks!

**Comment by lastzero** [comment]
Exit Code 143 indicates that the container successfully gracefully terminated after receiving the operating system's SIGTERM signal, which instructs the container to do so (otherwise you will see Exit Code 137).

**Comment by Moosbueffel** [comment]
Thanks that is reassuring,  but it doesn't really help. I don't want, that it stop every day.
And why does PhotoPrism reindex, when nothing changed?

**Comment by Moosbueffel** [comment]
I'm always fighting with this problem. Can anyone help me to find the source and to get it fixed?
Where can I start? Is it a problem of my container? I found a hint to restart automatically, but I think it will be only a workaround and not a real solution - isn't it?

Thanks

**Reply by lastzero** [reply]
Since your Pi only seems to have 2GB of physical memory, have you tried increasing the swap space to 6GB (and checking if it is active and no other limit is configured) and/or setting one of the config options for resource-constrained systems like `PHOTOPRISM_RESOLUTION_LIMIT` or `PHOTOPRISM_WORKERS`?

See also: 
- https://docs.photoprism.app/getting-started/troubleshooting/#fatal-server-errors
- https://docs.photoprism.app/getting-started/troubleshooting/performance/#troubleshooting
- https://docs.photoprism.app/getting-started/config-options/#storage
- https://docs.photoprism.app/getting-started/config-options/#index-workers

**Comment by Moosbueffel** [comment]
Thanks, I will check your links. Also I will update to a Raspi with 4 MB next time....
Currently I have 4 MB swap, because in your recommendation you wrote about 4 MB RAM, so I thought 6 MB with swap included will work. Also I look for RAM-consumption in a graph and it's only 50% used.

> free -h
>               gesamt       benutzt     frei      gemns.  Puffer/Cache verfügbar
> Speicher:      1,8Gi       862Mi    88Mi       6,0Mi       896Mi       914Mi
> Swap:          4,0Gi       1,7Gi       2,3Gi

Sorry, ich bekomme das nicht formatiert. Dachte eigentlich mit "code" sollte es passen, aber da stimmt's gar nicht - kein Zeilenumbruch

I'll report...

**Reply by lastzero** [reply]
We recommend 4 GB of swap space, but the *System Requirements* section in our documentation also states that 4 GB of physical memory is recommended:
- https://docs.photoprism.app/getting-started/raspberry-pi/#system-requirements

We may change the recommendation to use more, but especially if you use a microSD card on a Pi, it would rather make sense to reduce the memory usage as described in our documentation and to make sure that no other applications are running at the same time:
- https://docs.photoprism.app/getting-started/raspberry-pi/#troubleshooting

**Reply by lastzero** [reply]
Also see the following note:
> Indexing RAW images and high-resolution panoramas may require additional [swap space](https://docs.photoprism.app/getting-started/troubleshooting/docker/#adding-swap) and/or physical memory beyond the recommended minimum
